### DISCUSSION TITLE
AOT(Ahead-of-Time) compilation should be possible for GDScript.

### DISCUSSION THREAD

**Discussion by Foxchandaisuki** [godotengine/godot-proposals#9503] [Scripting]
I know we already have GDExtensions, and even C# code could be compiled to binary thanks to nativeAOT feature.

But languages for GDExtensions like C++, and C# have much more complex grammar than GDScript.

And they also lack so many learning materials that GDScript already has and will have.

In this condition, the best way to provide possibility to make their game faster than now for Godot junior developers is making GDScript compilable to Native binary code.

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot-proposals/issues/3069
* https://github.com/godotengine/godot-proposals/issues/8605

**Reply by Foxchandaisuki** [reply]
But they are not like C# NativeAOT of Godot engine and IL2CPP of Unity engine.

**Reply by AThousandShips** [reply]
What about those things are different and what about those things are necessary? I.e. why does it have to be just like those?

**Comment by Foxchandaisuki** [comment]
For convenience. For user, Typing GDScript and it becomes native binary
with no more action done by himself/herself is the best.

2024년 4월 11일 (목) 오후 9:55, A Thousand Ships ***@***.***>님이 작성:

**Reply by AThousandShips** [reply]
That's what those solutions suggest or offer though, that's just what AOT is, AOT just means you compile it ahead of time, not any specific fancy features (distinct from things like JIT)
